thriftclient/client.go

```diff
@@ -0,0 +1,62 @@
+package thriftclient
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// MonitoredClient implements the thrift.TClient interface and can be used to
+// wrap thrift calls in a client span.
+//
+// Calls will only be wrapped in client spans if the given context has a parent
+// span (either a server or "active" local/client span) set on it.  If no such
+// span is set, then the call will still proceed, it will just not be monitored
+// by a client span.
+type MonitoredClient struct {
+	client thrift.TClient
+}
+
+// NewMonitoredClientFromFactory returns a pointer to a new MonitoredClient that
+// a new TStandardClient using the provided transport and protocol factory.
+func NewMonitoredClientFromFactory(transport thrift.TTransport, factory thrift.TProtocolFactory) *MonitoredClient {
+	return NewMonitoredClient(thrift.NewTStandardClient(
+		factory.GetProtocol(transport),
+		factory.GetProtocol(transport),
+	))
+}
+
+// NewMonitoredClient returns a pointer to a new MonitoredClient that wraps the
+// given thrift.TClient calls in thrift client spans.
+func NewMonitoredClient(c thrift.TClient) *MonitoredClient {
+	return &MonitoredClient{client: c}
+}
```

Comment:
🔕 Do you think it's better to just make `client` an exported field, so we no longer need this constructor, and we can just rename the `NewMonitoredClientFromFactory` constructor to this shorter name? I think in most of the cases we want people to use that constructor anyways.

Reply:
That's probably fine?  I _like_ keeping thing un-exported but I probably over-favor that compared to how most go code seems to be.